onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -5649,12 +5649,87 @@ def aten_rnn_tanh_cell(
     raise NotImplementedError()
 
 
+@torch_op("aten::roll", trace_only=True)
 def aten_roll(
-    self: TensorType, shifts: Sequence[int], dims: Optional[Sequence[int]] = None
-) -> TensorType:
+    self: TTensor, shifts: Sequence[int], dims: Optional[Sequence[int]] = None
+) -> TTensor:
     """roll(Tensor self, int[1] shifts, int[1] dims=[]) -> Tensor"""
 
-    raise NotImplementedError()
+    self_rank = len(self.shape)
+    if self_rank == 0:
+        return self
+    elif self.shape[0] == 0:  # empty tensor
+        return self
+    else:
+        if dims is None:
```

Comment:
https://pytorch.org/docs/2.0/generated/torch.roll.html?highlight=roll#torch.roll
It is int or tuple of ints.

Reply:
Since this documentation documents the torch.* operator, not the aten operator, it is possible that they differ in some ways. The aten op is usually more constrained and we will only need to implement as the aten signature specifies.

Reply:
Probably for aten op, we should go to [here](https://github.com/pytorch/pytorch/blob/afd955f3de4050e9617245fbd40769859b10515b/aten/src/ATen/native/native_functions.yaml#L5834C70-L5834C70) for the correct signature, right? The default value of dims is [] which is empty.